### PR TITLE
P2P - Copy Ad - Buy - Float Rate

### PR DESCRIPTION
**Pull request by usman-deriv**
[ClickUp Card](https://app.clickup.com/t/20696747/P2PS-2840)

New Spec File: copyAdFloatRateBuy.cy.js
image

Function **c_getExistingAdDetailsForValidation** changed that was being used by other spec files, results: 
1. cypress/e2e/smoke/P2P/copyAdFixedRateBuy.cy.js
image

2. cypress/e2e/smoke/P2P/copyAdFixedRateSell.cy.js
image

### CODE REVIEW

File: cypress/support/commands/p2p.js [hunk 1]
```diff
@@ -268,42 +268,61 @@ Cypress.Commands.add(
...
+      .then((offerAmount) => {
+        sessionStorage.setItem('c_offerAmount', offerAmount)
+      })
+    if (rateType == 'fixed') {
+      cy.findByTestId('fixed_rate_type')
+        .invoke('val')
+        .then((rateValue) => {
+          sessionStorage.setItem('c_rateValue', rateValue.trim())
+        })
+    } else if (rateType == 'float') {
```
Comment (indulekha-deriv) [inline cypress/support/commands/p2p.js:287]: For rateType - Float isn't th code perfomaing same action for both adType - buy & adType - sell?

File: cypress/e2e/full/P2P/copyAdFloatRateBuy.cy.js
Comment (indulekha-deriv) [inline cypress/e2e/full/P2P/copyAdFloatRateBuy.cy.js:1]: In test case bank, subtask - 7 it asks to validate Payment methods and Counterparty Conditions.
Payment methods - Should not be able to edit/add/delete. 
Counterparty Conditions - Should not be able to edit - only view
are we covering this?

Comment (usman-deriv) [reply cypress/e2e/full/P2P/copyAdFloatRateBuy.cy.js:1]: This negative scenario is covered now. Thank you.

File: cypress/support/commands/p2p.js (hunk 1 above)
Comment (usman-deriv) [reply cypress/support/commands/p2p.js:288]: Same action is being performed for Buy and Sell Ad type. The only difference is the rate type (i.e. fixed and float) due to which the field for the ad rate differs. Other than this, the function is optimized enough to cater another spec for Float-Sell ad without making any changes to this function. PR -> https://github.com/deriv-com/e2e-deriv-app/pull/502

**Review by maliheh-deriv — APPROVED** [review]

**Review by indulekha-deriv — APPROVED** [review]

**Pull request merged**